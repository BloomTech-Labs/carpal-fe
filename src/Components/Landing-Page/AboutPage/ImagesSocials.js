import React from "react";

const ImagesSocials = (props) => {
    return (
        <>
            <div className="img-container" data-testid="img-container">
                <img
                    className="over"
                    src={props.carpaller.cutieImg}
                    alt="CarPal cutie"
                />
                <img
                    className="under"
                    src={props.carpaller.profileImg}
                    alt={props.carpaller.name}
                />
            </div>

            <div className="socials" data-testid="socials">
                <a
                    href={props.carpaller.githubURL}
                    target="_blank"
                    data-testid="github-url"
                >
                    <i className="fab fa-github fa-3x"></i>
                </a>
                <a href={props.carpaller.linkedinURL} target="_blank">
                    <i className="fab fa-linkedin-in fa-3x"></i>
                </a>
            </div>
        </>
    );
};

export default ImagesSocials;
