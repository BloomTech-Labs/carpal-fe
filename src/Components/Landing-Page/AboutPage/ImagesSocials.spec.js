import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import ImagesSocials from "./ImagesSocials";

afterEach(rtl.cleanup);

let props = {
    carpaller: {
        name: "Dennis Padiernos",
        title: "Chief Dad Joke Officer",
        cutieImg: "blueCutie.png",
        profileImg: "dennis.png",
        githubURL: "https://github.com/depadiernos",
        linkedinURL: "https://www.linkedin.com/in/dennispadiernos",
        bio:
            "Dennis is a software engineer who also loves music and mechanical keyboards and in his free time, he likes long walks by the beach and being a secret agent of OWCA (Organization Without a Cool Acronym)"
    }
};

describe("about card renders without crashing", () => {
    test("<ImagesSocials /> snapshot", () => {
        const wrapper = rtl.render(
            <Router>
                <ImagesSocials {...props} />
            </Router>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test("structure displays properly", () => {
        const { getByTestId } = rtl.render(
            <Router>
                <ImagesSocials {...props} />
            </Router>
        );
        expect(getByTestId("img-container")).toBeVisible();
        expect(getByTestId("socials")).toBeVisible();
        expect(getByTestId("github-url").getAttribute("href")).toBe(
            "https://github.com/depadiernos"
        );
    });
});
