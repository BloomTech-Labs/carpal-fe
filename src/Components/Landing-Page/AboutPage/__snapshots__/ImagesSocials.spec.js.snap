// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about card renders without crashing <ImagesSocials /> snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="img-container"
        data-testid="img-container"
      >
        <img
          alt="CarPal cutie"
          class="over"
          src="blueCutie.png"
        />
        <img
          alt="Dennis Padiernos"
          class="under"
          src="dennis.png"
        />
      </div>
      <div
        class="socials"
        data-testid="socials"
      >
        <a
          data-testid="github-url"
          href="https://github.com/depadiernos"
          target="_blank"
        >
          <i
            class="fab fa-github fa-3x"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/dennispadiernos"
          target="_blank"
        >
          <i
            class="fab fa-linkedin-in fa-3x"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="img-container"
      data-testid="img-container"
    >
      <img
        alt="CarPal cutie"
        class="over"
        src="blueCutie.png"
      />
      <img
        alt="Dennis Padiernos"
        class="under"
        src="dennis.png"
      />
    </div>
    <div
      class="socials"
      data-testid="socials"
    >
      <a
        data-testid="github-url"
        href="https://github.com/depadiernos"
        target="_blank"
      >
        <i
          class="fab fa-github fa-3x"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/dennispadiernos"
        target="_blank"
      >
        <i
          class="fab fa-linkedin-in fa-3x"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
